server/ladder_service/ladder_service.py

```diff
@@ -673,6 +708,35 @@ async def launch_match(
                 if player not in connected_players
             ])
 
+    def calculate_dynamic_tokens_per_map(self, M: float, tokens: list[int]) -> float:
```

Comment:
calculate_dynamic_tokens_per_map found the minimal possible max_tokens_per_map but while still respecting M

Reply:
but when you decide that `max_tokens_per_map` can't be reduced anymore?

Reply:
Its just some math
it solves series of equasions, if vetoes set correctly, one of them is guaranteed to be correct answer
if vetoes set incorrectly, it will return 0

Reply:
i'll prepare some explanation how math works